File: frontend/src/api/kutirs.ts
import api from "./client";

export interface Kutir {
  id: number; name: string; kutir_type: string; cluster_id: number;
  district_id: number | null; village: string | null; street: string | null;
  state: string; pincode: string | null; teacher_id: number | null;
  donor_id: number | null; enrollment_5th: number | null; enrollment_8th: number | null;
}
export interface KutirCreate {
  name: string; kutir_type: string; cluster_id: number; district_id?: number | null;
  village?: string | null; street?: string | null; state?: string; pincode?: string | null;
  enrollment_5th?: number | null; enrollment_8th?: number | null;
}
export const listKutirs = async (params?: { cluster_id?: number; district_id?: number }): Promise<Kutir[]> =>
  (await api.get("/kutirs", { params })).data;
export const createKutir = async (data: KutirCreate): Promise<Kutir> => (await api.post("/kutirs", data)).data;
export const updateKutir = async (id: number, data: Partial<KutirCreate>): Promise<Kutir> => (await api.put(`/kutirs/${id}`, data)).data;

File: backend/app/schemas/kutirs.py
from typing import Optional
from pydantic import BaseModel

KUTIR_TYPES = ["Seva Kutir", "Shiksha Kutir", "Non-Kutir"]
STATES = ["Madhya Pradesh", "Jharkhand", "Chhattisgarh"]

class KutirBase(BaseModel):
    name: str
    kutir_type: str = "Seva Kutir"
    cluster_id: int
    district_id: Optional[int] = None
    village: Optional[str] = None
    street: Optional[str] = None
    state: str = "Madhya Pradesh"
    pincode: Optional[str] = None
    teacher_id: Optional[int] = None
    donor_id: Optional[int] = None
    enrollment_5th: Optional[int] = None
    enrollment_8th: Optional[int] = None

class KutirCreate(KutirBase):
    pass

class KutirUpdate(BaseModel):
    name: Optional[str] = None
    kutir_type: Optional[str] = None
    cluster_id: Optional[int] = None
    district_id: Optional[int] = None
    village: Optional[str] = None
    street: Optional[str] = None
    state: Optional[str] = None
    pincode: Optional[str] = None
    teacher_id: Optional[int] = None
    donor_id: Optional[int] = None
    enrollment_5th: Optional[int] = None
    enrollment_8th: Optional[int] = None

class KutirOut(KutirBase):
    id: int
    model_config = {"from_attributes": True}
